iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/SetGeneratorToLocalRegulation.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.modification;
+
+import com.powsybl.commons.report.ReportNode;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.topology.NamingStrategy;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Network;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+import static com.powsybl.iidm.modification.util.ModificationReports.generatorLocalRegulationReport;
+
+/**
+ * <p>Network modification to set a generator regulation to local instead of remote.</p>
+ * <ul>
+ *     <li>Generator's RegulatingTerminal is set to the generator's own Terminal.</li>
+ *     <li>TargetV engineering unit value is adapted but the per unit value remains the same.</li>
+ * </ul>
+ *
+ * @author Romain Courtier {@literal <romain.courtier at rte-france.com>}
+ */
+public class SetGeneratorToLocalRegulation extends AbstractNetworkModification {
+
+    private final String generatorId;
+    private static final Logger LOG = LoggerFactory.getLogger(SetGeneratorToLocalRegulation.class);
+
+    public SetGeneratorToLocalRegulation(String generatorId) {
+        this.generatorId = Objects.requireNonNull(generatorId);
+    }
+
+    @Override
+    public String getName() {
+        return "SetGeneratorToLocalRegulation";
+    }
+
+    @Override
+    public void apply(Network network, NamingStrategy namingStrategy, boolean throwException, ComputationManager computationManager, ReportNode reportNode) {
+        Generator generator = network.getGenerator(generatorId);
+        if (generator != null
+                && generator.getTerminal() != null
+                && generator.getRegulatingTerminal() != null
+                && generator.getTerminal() != generator.getRegulatingTerminal()) {
```

Comment:
Tests on references may fail depending on the used IIDM implementation. To avoid problems, these tests should be done using IDs. In this case, you could use the terminal's connectable ID.

Reply:
Done.
I have also removed the ```generator.getRegulatingTerminal() != null``` checks since the regulatingTerminal is never null (if not set at initialization or when explicitely attempting to set it to null, it is always reset to the generator's own terminal)